File: omega up/palabra corta.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int num_palabras;

    //cout << "Ingresa el numero de palabras: ";
    cin >> num_palabras;
    cin.ignore(numeric_limits<streamsize>::max(), '\n'); // Consumir la nueva línea

    string palabra_corta = "";
    int longitud_corta = numeric_limits<int>::max(); // Valor máximo para la longitud inicial

    for (int i = 0; i < num_palabras; i++) {
        string palabra;
        //cout << "Ingresa tu palabra: ";
        getline(cin, palabra);

        if (palabra.length() < longitud_corta) {
            palabra_corta = palabra;
            longitud_corta = palabra.length();
        }
    }

    cout /*<< "La palabra más corta es: " */ << palabra_corta <<endl;
    cout /*<< "Tiene " */ << longitud_corta << /*" caracteres"*/endl;

    if (longitud_corta % 2 == 0) {
        cout << " par " << endl;
    } else {
        cout << " impar " << endl;
    }

    return 0;
}

File: omega up/Palabra_Corta.java
package omegaup;
import java.util.*;

public class Palabra_Corta {
    public static void main(String[] args) {
        Scanner teclado = new Scanner(System.in);
        String Palabra;
        int num_palabras;

        System.out.println("Ingresa el numero de palabras");
        num_palabras = teclado.nextInt();
        teclado.nextLine();

        String palabra_corta = "";
        int longitud_corta = Integer.MAX_VALUE;

        for(int i = 0; i < num_palabras; i++){
            System.out.println("Ingresa tu palabra: ");
            Palabra = teclado.nextLine().trim();
            if(Palabra.length() < longitud_corta){
                palabra_corta = Palabra;
                longitud_corta = Palabra.length();
            }
        }
        System.out.println("La palabra más corta es: " + palabra_corta);
        System.out.println("Tiene " + longitud_corta + " caracteres");

        if(longitud_corta % 2 == 0){
            System.out.println("La longitud de la palabra es par");
        }else{
            System.out.println("La longitud de la palabra es impar");
        }
    }
}
